### PR TITLE
Misc. updates

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Copy edits stashed from recent PRs

### CODE REVIEW

File: content/docs/command-reference/add.md
Comment (shcheklein) [inline content/docs/command-reference/add.md:146]: hmm, not sure this is true/relevant here

File: content/docs/command-reference/add.md
Comment (shcheklein) [inline content/docs/command-reference/add.md:149]: also, looks like `--glob` is out of order with the other options?

File: content/docs/command-reference/add.md
Comment (jorgeorpinel) [reply content/docs/command-reference/add.md:146]: This was added after noticing people don't realize --external implies --no-commit but I don't have the links to those conversations right now. I only remember the last one (somewhere on Discord): basically a user asking how to backup external outputs.

So when(/if?) the import --store option gets merged, this will be revisited anyway. Up to you. I think that if it's not wrong, we should err on the side of TMI (too much info) in cmd refs.

File: content/docs/command-reference/add.md
Comment (jorgeorpinel) [reply content/docs/command-reference/add.md:149]: I guess it can go before --external? I'm not sure which one we expect people to use more. I moved it up one place for now.

File: content/docs/command-reference/add.md
Comment (shcheklein) [reply content/docs/command-reference/add.md:146]: It looks wrong to me that the biggest issue. Unless I'm missing something.

File: content/docs/command-reference/add.md
Comment (shcheklein) [reply content/docs/command-reference/add.md:149]: it should be in the same order as DVC CLI has it in, right?

Comment (jorgeorpinel) [reply content/docs/command-reference/add.md]: Not necessarily. We use an arbitrary order here. The core team tends to always append options at the end. There's no consistency so far 🙁.

There's a discussion about this in iterative/dvc/issues/3642.

File: content/docs/command-reference/add.md
Comment (jorgeorpinel) [reply content/docs/command-reference/add.md:150]: It's probably not a big issue. Thus it's a block quote note. So, should we remove it?

File: content/docs/command-reference/add.md
Comment (shcheklein) [reply content/docs/command-reference/add.md]: kk

File: content/docs/command-reference/add.md
Comment (shcheklein) [reply content/docs/command-reference/add.md:150]: It's kinda an issue if this information is wrong. I'm fine with both - remove and/or clarify and confirm.

Comment (jorgeorpinel) [reply content/docs/command-reference/add.md]: I guess you're right, external outputs can be cached, but externally. Is that what you meant? Sorry, just getting it now... Updated to:

  > Note that external outputs require an external cache setup (unless
  > `--no-commit` is used), and are never pushed or pulled from/to remote
  > storage. See link above for more details.

File: content/docs/command-reference/add.md
Comment (shcheklein) [inline content/docs/command-reference/add.md:151]: is there a reasonable workflow with `--no-commit` + `--external`?

Comment (jorgeorpinel) [reply content/docs/command-reference/add.md:151]: I think so, especially since we mention "For cached external outputs..." in https://dvc.org/doc/user-guide/managing-external-data

For example if you want to avoid the external data from being moved and linked back to its position for other processes that also use it which may not support handling links.

File: content/docs/command-reference/gc.md
Comment (jorgeorpinel) [inline content/docs/command-reference/gc.md:97]: Here

File: content/docs/command-reference/status.md
Comment (jorgeorpinel) [inline content/docs/command-reference/status.md:147]: And here

**Review by jorgeorpinel — COMMENTED** [review]
Note changes per https://github.com/iterative/dvc.org/pull/1941#discussion_r524541298:

File: content/docs/command-reference/add.md
Comment (shcheklein) [reply content/docs/command-reference/add.md:151]: okay, we don't move data, but we also don't calculate any hashes in this case, right? what would the next step for this `.dvc` file?

Comment (jorgeorpinel) [reply content/docs/command-reference/add.md:151]: I just tried this and it does calculate the hash. The data is just not moved to the cache. (At least using an external data file in the local system)

Comment (jorgeorpinel) [reply content/docs/command-reference/add.md:151]: And fortunately the --no-commit option description doesn't state that the has isn't calculated.

Unrelated: It does however mention the state file (fixed the path in 2ab35cc2) which is probably completely out of context/unnecessary to mention here but oh well...

Comment (shcheklein) [reply content/docs/command-reference/add.md:151]: `I just tried this and it does calculate the hash` - hmm looks like a bug to me. Local `--no-commit` avoids calculating hash for outputs as far as I remember. Let me check.

Comment (jorgeorpinel) [reply content/docs/command-reference/add.md:151]: It's calculating it even for regular outputs in the workspace for me (DVC 1.9.1 on Win).

Comment (shcheklein) [reply content/docs/command-reference/add.md:151]: okya, you are right on this (my usual confusion since --no-commit is not an opposite to `dvc commit` unfortunately)

it calculates hashes

```
√ Projects/test-dvc % dvc add --external --no-commit s3://dvc-temp/ivan/images
Adding...
ERROR: Current operation was unsuccessful because 's3://dvc-temp/ivan/images' requires existing cache on 's3' remote. See <https://man.dvc.org/config#cache> for information on how to set up remote cache.
```

it is still not correct it seems.

To be honest I doubt that this note could be understood by anyone :) I would keep this information to the Managing external data section.

Comment (shcheklein) [reply content/docs/command-reference/add.md:151]: > It does however mention the state file

it is irrelevant, agreed

Comment (jorgeorpinel) [reply content/docs/command-reference/add.md]: > I would keep this information to the Managing external data section
> it (state file) is irrelevant, agreed

Ok, updated both things in 7bad093e. Just left the note about the external cache, which still seems pretty relevant IMO.

Comment (jorgeorpinel) [reply content/docs/command-reference/add.md]: PTAL

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [inline content/docs/user-guide/managing-external-data.md:40]: Number of notes is getting out of hands. If we consider this important, let's improve the explanation how they work?

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [inline content/docs/user-guide/managing-external-data.md:25]: not cached and no-commit are different things (like -O and --no-commit for regular outputs) Also, as I checked `--no-commit` still requires cache to be set.

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md]: I removed the first one (also from the external deps doc for consistency) and made the 2nd one not a block quote. So now there's just one block quote left at the bottom.

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md:25]: OK... I removed the last sentence for now. It's kind of confusing/hard to explain right. Maybe we should review the behavior in the core repo first?

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [inline content/docs/user-guide/managing-external-data.md:36]: it's not about note literally. It's about text becoming a list of disconnected paragraphs. Question- why do we have this paragraph here, after the list of supported types? Can we make it organically part of the text above?

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [reply content/docs/user-guide/managing-external-data.md]: `--no-commit` workflow is not about caching/not caching. Usually it is about deliberately _delaying_ the caching for some reason.

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md:36]: Hmmm, I'd prefer not to mix the previous paragraph with the concept of "remote storage". That will add confusion. I think that's why these mentions of remote storage are after the bullet list. I could possibly mix them into a more narrative paragraph (instead of descriptive notes)... ⌛

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md]: OK. Is that relevant to mention especially in this guide? Seems like a general thing for add/run (and it's already mentioned in those refs).

The reason it was mentioned here is that it is the exception to the "Normally" at the beginning of the paragraph.

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [reply content/docs/user-guide/managing-external-data.md:36]: it's not about the previous paragraph literally. It's about making it part of the some text above. Probably the one about `cache` works good. It can explain why does it happen. And even note about hash collision (I hope it won't be relevant) should go probably before the list of supported remote types.

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [reply content/docs/user-guide/managing-external-data.md]: Yep, I would avoid changing it to `Normally, ` and then we don't need mentioning this.

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md:36]: Yes, you didn't even mention the previous paragraph. I was also trying to refer to all the text before the list. I think mentioning remote storage in most places above may be confusing but let me try ⌛

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md]: OK, I'm removing "Normally", but then this isn't always true. But maybe it will be fixed by the changes per our other discussion here ⌛

Comment (shcheklein) [reply content/docs/user-guide/managing-external-data.md]: They do always require the external cache as we checked, right? Also, I don't see a good use case w/o cache.

Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md]: Yes they do. Which seems like a bug to me, but if there's no use case for the combination anyway then I guess this is not an issue. Resolving!

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md]: OK I mostly managed to do this, while also improving the explanations a little, I think. But I did have to leave one block quote note under the list of remote types (in the ext deps doc). PTAL @shcheklein

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [inline content/docs/user-guide/managing-external-data.md:42]: Note that I totally removed this note. How likely is this scenario really? And if we're worried about it, should DVC core handle it as an error, at least?

Comment (shcheklein) [reply content/docs/user-guide/managing-external-data.md:42]: It's hard for DVC to handle this. It will change a bit in DVC 2.0, so it might be safe to remove it. On the other hand, it's a nature of clouds- they generate some hash that might differ from the one we generate with DVC. And there is no any guarantees. The best we can do is to have per cloud instruction- is it safe or not mixing those things.

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [inline content/docs/user-guide/managing-external-data.md:19]: it's not only about detecting changes, it's about versioning them (that's where we use "track")

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [inline content/docs/user-guide/managing-external-data.md:35]: I'm a bit lost here ... I think the previous version was pretty precise, why change it?

File: content/docs/user-guide/managing-external-data.md
Comment (efiop) [reply content/docs/user-guide/managing-external-data.md:42]: I would keep it for now. Don't see a good reason to remove it.

Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md:42]: > Don't see a good reason to remove it.

Doing so per https://github.com/iterative/dvc.org/pull/1945#pullrequestreview-531982134 and https://github.com/iterative/dvc.org/pull/1945#pullrequestreview-531992838 above. And since it seems to me like an extremely rare edge case. And if it will change in 2.0 we can think about what to say about it then.

Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md:42]: > It's hard for DVC to handle this.

BTW if the problem is "hash overlaps: the hash of an external output could collide with a hash generated locally for another file" that sounds like a very precise condition that could be caught/handled. But I'm not sure.

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md:35]: > why change it?

The previous version was:

> For cached external outputs (e.g. dvc add, dvc run -o), you will need to setup an external cache in the same external/remote file system first.

We decided to not mentioned "cached outs" here since there's no real use case for non cached external outs. The mention of `add` and `run -o` also don't seem that important. So most of that sentence had to go anyway. Then we decided to incorporate the info involving remotes to the paragraphs before the bullet list so pretty much all the text changed. And it was a good thing, I think, because it explains what ext outs are first, emphasizes in dvc.yaml, and how their URLs are formed (same as remotes).

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md:19]: Good point. But "track when they change" doesn't sound right... I went for "track them for [versioning], thus detecting when they change".

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md]: I removed "Importantly" now, and moved this paragraph before the note about remotes being a different feature (right after the bullet list). I think the structure of this section makes sense: 1 what are ext outs, 2 how to define them, 3 note about external cache.

File: content/docs/user-guide/managing-external-data.md
Comment (efiop) [reply content/docs/user-guide/managing-external-data.md:42]: @jorgeorpinel There is no guarantee it will go away in 2.0. This note discourages people from misusing our incomplete ext feature, so better to not remove it for now.

Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md:42]: OK sure, I rolled this back (mostly) in defd99d8 @efiop.

What did you mean by "incomplete ext feature" thouogh? ext = external data? Just curious

File: content/docs/user-guide/external-dependencies.md
Comment (shcheklein) [inline content/docs/user-guide/external-dependencies.md:5]: `even if it's external or remote to the project` - no need for this. It complicates it.

File: content/docs/user-guide/external-dependencies.md
Comment (shcheklein) [inline content/docs/user-guide/external-dependencies.md:17]: to be honest mentioning `dvc status` is strange here. The most important thing that it will for example detect this change on `dvc repro`.

File: content/docs/user-guide/external-dependencies.md
Comment (shcheklein) [inline content/docs/user-guide/external-dependencies.md:31]: why is it needed?

File: content/docs/user-guide/external-dependencies.md
Comment (shcheklein) [inline content/docs/user-guide/external-dependencies.md:35]: let's simplify this ... reading two paragraphs of text as intro to an example looks wrong

File: content/docs/user-guide/external-dependencies.md
Comment (shcheklein) [inline content/docs/user-guide/external-dependencies.md:42]: this note is also not needed here I think

File: content/docs/user-guide/external-dependencies.md
Comment (shcheklein) [reply content/docs/user-guide/external-dependencies.md:35]: I think it was fine to keep in the previous paragraph?

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [inline content/docs/user-guide/managing-external-data.md:5]: same here

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [inline content/docs/user-guide/managing-external-data.md:16]: DVC pipelines stages?

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [inline content/docs/user-guide/managing-external-data.md:38]: same here, why is it needed?

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [inline content/docs/user-guide/managing-external-data.md:53]: as an external cache?

File: content/docs/user-guide/managing-external-data.md
Comment (shcheklein) [inline content/docs/user-guide/managing-external-data.md:54]: on the location - sounds strange a bit

File: content/docs/user-guide/external-dependencies.md
Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md]: Removed in 8a6892dd.

File: content/docs/user-guide/external-dependencies.md
Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md]: Good point. I changed these texts in dd1501bb for both external docs since describing what deps/outs are isn't a goal here.

File: content/docs/user-guide/external-dependencies.md
Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md]: Because we mention remotes before so it may be confusing. Is a short note by the end of the section too much?

File: content/docs/user-guide/external-dependencies.md
Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md]: > it was fine to keep in the previous paragraph

Yes, going back to that (almost) now that the mechanism is explained a little more in the previous section.

Done in de67413d

File: content/docs/user-guide/external-dependencies.md
Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md:42]: Gone.

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md]: Not seeing need to mention pipelines in these docs, we're already going over deps/outs and stages. Too many concepts otherwise, probably? You can click on "stage" to get to pipelines.

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md:52]: No "an" needed. Maybe "the" but it's OK like this too, I think.

File: content/docs/user-guide/managing-external-data.md
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md:53]: Yes. Going for "on the external location". Thanks

File: content/docs/user-guide/external-dependencies.md
Comment (shcheklein) [inline content/docs/user-guide/external-dependencies.md:10]: track and version?

File: content/docs/user-guide/external-dependencies.md
Comment (shcheklein) [inline content/docs/user-guide/external-dependencies.md:19]: sounds like it repeats a lot of the previous paragraph?

File: content/docs/user-guide/external-dependencies.md
Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md:10]: Do we version dependencies? Only when they're outputs of a previous stage, I think?

Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md:10]: I guess we do keep track of their versions in any case, but don't really control those versions if they are external.

**Review by shcheklein — APPROVED** [review]
looks good to me, a few minor questions

File: content/docs/user-guide/external-dependencies.md
Comment (shcheklein) [reply content/docs/user-guide/external-dependencies.md:10]: This sentence is about `External dependencies and [external outputs]`

File: content/docs/user-guide/external-dependencies.md
Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md:19]: True 🤦. Updated in cfe6e07a.

**Comment by jorgeorpinel** [comment]
Merging this! (I have lots of staged regular updates 😅) but please follow up on any discussions above @shcheklein, and I'll address them in a coming up PR.

File: content/docs/user-guide/external-dependencies.md
Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md:10]: True. Updating in 60e8055d.

**Pull request merged**
